pkg/operator/targetconfigcontroller/targetconfigcontroller_test.go

```diff
@@ -88,14 +96,45 @@ func TestTargetConfigController(t *testing.T) {
 				u.FakeEtcdMemberWithoutServer(0),
 				u.FakeEtcdMemberWithoutServer(2),
 			},
+			etcdBackupSpec:    nil,
 			etcdMembersEnvVar: "1,3",
 		},
+		{
+			name: "Backup Var Test",
+			objects: []runtime.Object{
+				u.BootstrapConfigMap(u.WithBootstrapStatus("complete")),
+			},
+			staticPodStatus: u.StaticPodOperatorStatus(
+				u.WithLatestRevision(3),
+				u.WithNodeStatusAtCurrentRevision(3),
+				u.WithNodeStatusAtCurrentRevision(3),
+				u.WithNodeStatusAtCurrentRevision(3),
+			),
+			etcdMembers: []*etcdserverpb.Member{
+				u.FakeEtcdMemberWithoutServer(0),
+				u.FakeEtcdMemberWithoutServer(1),
+				u.FakeEtcdMemberWithoutServer(2),
+			},
+			etcdBackupSpec:    u.CreateEtcdBackupSpecPtr("GMT", "0 */2 * * *"),
+			etcdMembersEnvVar: "1,2,3",
+		},
 	}
 	for _, scenario := range scenarios {
 		t.Run(scenario.name, func(t *testing.T) {
-			eventRecorder, _, controller := getController(t, scenario.staticPodStatus, scenario.objects, scenario.etcdMembers)
+			backupVar := backuphelpers.NewDisabledBackupConfig()
+			eventRecorder, _, controller, fakeKubeClient := getController(t, scenario.staticPodStatus, scenario.objects, scenario.etcdMembers, backupVar)
+			if scenario.etcdBackupSpec != nil {
+				backupVar.SetBackupSpec(scenario.etcdBackupSpec)
+			}
 			err := controller.sync(context.TODO(), factory.NewSyncContext("test", eventRecorder))
 			assert.Equal(t, scenario.expectedErr, err)
+
+			if scenario.etcdBackupSpec != nil {
```

Comment:
if it's nil, we should also assert the "--enabled=false" path here

Reply:
added, however this broke another case (i.e. `Quorum not tolerant`, since there was `CM[etcdpod.yaml]` to assert

Therefore this assertion only takes place iff the scenario has no error

Reply:
yes, because in that case the controller has no output. Can we only run the assertions if the "etcdpod.yaml" exists in the configmap instead?

Reply:
I'll send a suggestion, one sec, had a caching issue

Reply:
done 👍🏽